lib/EnsEMBL/REST/Model/ga4gh/Beacon.pm

```diff
@@ -362,22 +354,29 @@ sub beacon_query {
     $input_coord{'ref_allele'} = $data->{referenceBases};
     $input_coord{'alt_allele'} = $data->{alternateBases} ? $data->{alternateBases} : $data->{variantType};
 
-    # SNV or small indels have a start
-    # Structural variants can have start-end or startMin/startMax-endMin/endMax
-    $input_coord{'start'} = $data->{start} ? $data->{start} : $data->{startMin};
-    $input_coord{'start_max'} = $data->{startMax} ? $data->{startMax} : $input_coord{'start'};
+    # Types of query:
+    #  1) 'start' with no 'end'; can be done for SNV and small indels
+    #  2) 'start' and 'end' (range query); any variant falling fully or partially within the range
+    #  3) [start1, start2] and [end1, end2] (bracket query); can be used to match any contiguous genomic interval, e.g. for querying imprecise positions
+    my @start_list = split /,/, $data->{start};
+    my @end_list = split /,/, $data->{end} if ($data->{end});
+    my $start_1 = $start_list[0] if($start_list[0]);
+    my $start_2 = $start_list[1] if($start_list[1]);
+    my $end_1 = $end_list[0] if($end_list[0]);
+    my $end_2 = $end_list[1] if($end_list[1]);
+
+    $input_coord{'start'} = $start_1;
+    $input_coord{'start_max'} = $start_2 ? $start_2 : $input_coord{'start'};
 
     my $end;
     if($data->{end}) {
-      $input_coord{'end'} = $data->{end};
+      $input_coord{'end'} = $end_2 ? $end_2 : $end_1;
+      $input_coord{'end_min'} = $end_2 ? $end_1 : $input_coord{'end'};
     }
-    elsif($data->{endMax}) {
-      $input_coord{'end'} = $data->{endMax};
-    }
-    else{
+    else {
       $input_coord{'end'} = $input_coord{'start'};
+      $input_coord{'end_min'} = $input_coord{'start'};
```

Comment:
Should we support a query with `[start1, start2]` but only one `end`? If so, would it make sense to set `end_min` the same as `start_max`?

According to the docs, I don't think that this should occur, but just wanted to confirm: http://docs.genomebeacons.org/variant-queries/#term-use-comparison

```suggestion
      $input_coord{'end'} = $input_coord{'start'};
      $input_coord{'end_min'} = $input_coord{'start_max'};
```

Reply:
It is more correct to use `start_max`. Going to update.